test/wasi/test-return-on-exit.js

```diff
@@ -0,0 +1,18 @@
+// Flags: --experimental-wasi-unstable-preview1 --experimental-wasm-bigint
+'use strict';
+require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+const { WASI } = require('wasi');
+const wasi = new WASI({ returnOnExit: true });
+const importObject = { wasi_snapshot_preview1: wasi.wasiImport };
+const wasmDir = path.join(__dirname, 'wasm');
+const modulePath = path.join(wasmDir, 'exitcode.wasm');
+const buffer = fs.readFileSync(modulePath);
+
+(async () => {
+  const { instance } = await WebAssembly.instantiate(buffer, importObject);
+
+  assert.strictEqual(wasi.start(instance), 120);
+})();
```

Comment:
```suggestion
})().then(common.mustCall());
```

(I know other tests would fail if this was broken, too, but I think we should use this consistently for async functions)